main.tf

```diff
@@ -130,24 +115,41 @@ module "eks" {
   deploy_id                 = var.deploy_id
   private_subnets           = local.private_subnets
   ssh_pvt_key_path          = aws_key_pair.domino.key_name
-  route53_hosted_zone_name  = var.route53_hosted_zone_name
   bastion_security_group_id = try(module.bastion[0].security_group_id, "")
   create_bastion_sg         = var.create_bastion
   kubeconfig_path           = local.kubeconfig_path
   default_node_groups       = var.default_node_groups
   additional_node_groups    = var.additional_node_groups
-  s3_buckets                = module.storage.s3_buckets
+}
+
+module "storage" {
+  source                       = "./submodules/storage"
+  deploy_id                    = var.deploy_id
+  efs_access_point_path        = var.efs_access_point_path
+  s3_force_destroy_on_deletion = var.s3_force_destroy_on_deletion
+  subnets = [for s in local.private_subnets : {
+    name       = s.name
+    id         = s.id
+    cidr_block = s.cidr_block
+  }]
+  vpc_id = module.network.vpc_id
+  roles  = module.eks.eks_node_roles
+}
+
+data "aws_iam_role" "eks_master_roles" {
+  for_each = toset(var.eks_master_role_names)
+  name     = each.key
 }
 
 module "k8s_setup" {
-  source                  = "./submodules/k8s"
-  ssh_pvt_key_path        = abspath(local.ssh_pvt_key_path)
-  bastion_user            = local.bastion_user
-  bastion_public_ip       = try(module.bastion[0].public_ip, "")
-  k8s_cluster_endpoint    = module.eks.cluster_endpoint
-  managed_nodes_role_arns = module.eks.managed_nodes_role_arns
-  eks_master_role_names   = concat(var.eks_master_role_names, module.eks.eks_master_role_name)
-  kubeconfig_path         = local.kubeconfig_path
+  source               = "./submodules/k8s"
+  ssh_pvt_key_path     = abspath(local.ssh_pvt_key_path)
+  bastion_user         = local.bastion_user
+  bastion_public_ip    = try(module.bastion[0].public_ip, "")
+  k8s_cluster_endpoint = module.eks.cluster_endpoint
+  eks_node_role_arns   = [for r in module.eks.eks_node_roles : r.arn]
```

Comment:
Either the var is not named correctly or its contents do not match,  i would expect this var to produce
```hcl
managed_nodes_role_arns = module.eks.managed_nodes_role_arns
```

Reply:
I don't want to make hard assumptions about what the upper layer require.
You originally return the names and then converted it back to arns.
Yes, I could return arns but it gets very messy to determine what is needed by other modules so let them decide the inputs and make the outputs pretty transparent.